File: backend/src/main/java/com/incidentcommand/dto/IncidentResponse.java
package com.incidentcommand.dto;

import com.incidentcommand.model.enums.IncidentStatus;
import com.incidentcommand.model.enums.IncidentType;
import com.incidentcommand.model.enums.Severity;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class IncidentResponse {
    private Long id;
    private String title;
    private String description;
    private Severity severity;
    private IncidentStatus status;
    private IncidentType type;
    private String assignedTeamName;
    private String assignedUsername;
    private String createdByUsername;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
    private LocalDateTime resolvedAt;
}

File: android/app/src/main/java/com/incidentcommand/android/data/remote/dto/IncidentResponse.kt
package com.incidentcommand.android.data.remote.dto

data class IncidentResponse(
    val id: Long,
    val title: String,
    val description: String?,
    val severity: String,
    val status: String,
    val type: String,
    val assignedTeamName: String?,
    val assignedUsername: String?,
    val createdByUsername: String,
    val createdAt: String,
    val updatedAt: String,
    val resolvedAt: String?
)
